### PR TITLE
MachineDeployment controller: expose errors

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Currently the machineDeployment controller only logs errors if someone happened to decide to log a given error which is sometimes the case and sometimes not. An event on the MachineDeployment is never created.

This PR changes the behaviour to always log an error and create an even when an error happened during reconciliation

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
none
```

/assign @vincepri

### CODE REVIEW

File: pkg/controller/machinedeployment/controller.go [hunk 1]
```diff
@@ -155,7 +162,16 @@ func (r *ReconcileMachineDeployment) Reconcile(request reconcile.Request) (recon
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
+	result, err := r.reconcile(ctx, d)
+	if err != nil {
+		klog.Errorf("failed to reconcile MachineDeployment %s: %v", request.NamespacedName, err)
```
Comment (vincepri) [inline pkg/controller/machinedeployment/controller.go]:
```suggestion
		klog.Errorf("Failed to reconcile MachineDeployment %q: %v", request.NamespacedName, err)
```

File: pkg/controller/machinedeployment/controller.go [hunk 2]
```diff
@@ -155,7 +162,16 @@ func (r *ReconcileMachineDeployment) Reconcile(request reconcile.Request) (recon
...
+		r.recorder.Eventf(d, corev1.EventTypeWarning, "ReconcilingError", "%v", err)
```
Comment (vincepri) [inline pkg/controller/machinedeployment/controller.go]:
```suggestion
		r.recorder.Eventf(d, corev1.EventTypeWarning, "ReconcileError", "%v", err)
```

File: pkg/controller/machinedeployment/controller.go [hunk 3]
```diff
@@ -43,15 +45,19 @@ var (
 	controllerKind = v1alpha1.SchemeGroupVersion.WithKind("MachineDeployment")
 )
 
+// controllerName is the name of this controller<Paste>
```
Comment (detiber) [inline pkg/controller/machinedeployment/controller.go]:
```suggestion
// controllerName is the name of this controller
```

**Comment by alvaroaleman** [comment]
@vincepri @detiber wow thanks for the quick review!

**Pull request merged**